bin/propolis-server/src/lib/stats.rs

```diff
@@ -10,31 +10,38 @@ use oximeter::{
     types::{Cumulative, ProducerRegistry, Sample},
     Metric, MetricsError, Producer,
 };
-use oximeter_producer::{Config, Error, Server};
-use slog::Logger;
-
-use std::net::SocketAddr;
-use std::sync::{Arc, Mutex};
-use uuid::Uuid;
-
-use crate::server::MetricsEndpointConfig;
-use crate::stats::virtual_machine::VirtualMachine;
-
 // Propolis is built and some tests are run on non-illumos systems. The real
 // `kstat` infrastructure cannot be built there, so some conditional compilation
 // tricks are needed
 #[cfg(all(not(test), target_os = "illumos"))]
 use oximeter_instruments::kstat::KstatSampler;
+use oximeter_producer::{self, Config, Server};
+use slog::Logger;
+use std::{
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
 
+use crate::server::MetricsEndpointConfig;
+
+mod network_interface;
 mod pvpanic;
-pub(crate) mod virtual_machine;
-pub use self::pvpanic::PvpanicProducer;
+mod virtual_machine;
+
+pub(crate) use self::network_interface::InstanceNetworkInterfaces;
+pub(crate) use self::pvpanic::PvpanicProducer;
+pub(crate) use self::virtual_machine::VirtualMachine;
 
 // Interval on which we ask `oximeter` to poll us for metric data.
 const OXIMETER_STAT_INTERVAL: tokio::time::Duration =
     tokio::time::Duration::from_secs(30);
 
-// Interval on which we produce vCPU metrics.
+/// Interval on which we sample instance/guest network interface metrics.
+#[cfg(all(not(test), target_os = "illumos"))]
+const NETWORK_INTERFACE_SAMPLE_INTERVAL: std::time::Duration =
+    std::time::Duration::from_secs(10);
```

Comment:
Why are the three different stat intervals different?

Reply:
They are different. The interface one is similar to what we do with sled metrics vs what was set for vcpu beforehand.

Reply:
@pfmooney `OXIMETER_STAT_INTERVAL` refers to the interval on which we ask `oximeter` to collect data from Propolis. The other intervals are used in the `KstatSampler` to determine how frequently to generate samples from the kstats themselves. Each item registered with the sampler defines its own sampling interval, since there's no one-size-fits-all.

I'm actually going to rename this for clarity in #746 to help address this.